src/site/content/en/blog/gamepad/index.md

```diff
@@ -0,0 +1,190 @@
+      Source
+    </a>).
+  </figcaption>
+</figure>
+
+### Being notified when a gamepad gets connected
+
+The way an app can find out if a gamepad is connected is by listening for the `gamepadconnected`
+event that triggers on the `window` object.
+When the user connects a gamepad, which can either happen via USB or via Bluetooth,
+a `GamepadEvent` is fired that has the gamepad's details in an aptly named `gamepad` property.
+Below you can see an example from an Xbox 360 controller that I had lying around
+(yes, I am into retro gaming).
+
+```js
+window.addEventListener('gamepadconnected', (event) => {
+  console.log('✅ 🎮 A gamepad was connected:', event.gamepad);
+  /*
+    gamepad: Gamepad
+    axes: (4) [0, 0, 0, 0]
+    buttons: (17) [GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton]
+    connected: true
+    id: "Xbox 360 Controller (STANDARD GAMEPAD Vendor: 045e Product: 028e)"
+    index: 0
+    mapping: "standard"
+    timestamp: 6563054.284999998
+    vibrationActuator: GamepadHapticActuator {type: "dual-rumble"}
+  */
+});
+```
+
+### Being notified when a gamepad gets disconnected
+
+Being notified of gamepad disconnects happens analogously to the way connections are detected.
+Just this time the app listens for the `gamepaddisconnect` event.
+Note how in the example below `connected` is now `false`.
+
+```js
+window.addEventListener('gamepadconnected', (event) => {
+  console.log('❌ 🎮 A gamepad was disconnected:', event.gamepad);
+  /*
+    gamepad: Gamepad
+    axes: (4) [0, 0, 0, 0]
+    buttons: (17) [GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton, GamepadButton]
+    connected: false
+    id: "Xbox 360 Controller (STANDARD GAMEPAD Vendor: 045e Product: 028e)"
+    index: 0
+    mapping: "standard"
+    timestamp: 6563054.284999998
+    vibrationActuator: null
+  */
+});
+```
+
+### The gamepad in your game loop
+
+Getting a hold of a gamepad starts with a call to `navigator.getGamepads()`,
+which returns a `GamepadList` object with `Gamepad` items.
+Note that for this to work, you may need to "wake" your gamepad by pressing any of its buttons.
+More than one gamepad can be connected to a device,
+so be sure to check all items of the `GamepadList`.
+
+```js
+const pollGamepad = () => {
+  // Always call `navigator.getGamepads()` inside of
+  // the game loop, not outside.
+  const gamepads = navigator.getGamepads();
+  for (const gamepad of gamepads) {
+    if (!gamepad) {
```

Comment:
Can you explain this line?
How could we get a falsy `gamepad` object from `gamepads`?

Reply:
```js
pads = navigator.getGamepads()
// GamepadList {0: null, 1: null, 2: null, 3: null, length: 4}
for (const pad of pads) { 
  console.log(pad)
}
/*
VM221:2 null
VM221:2 null
VM221:2 null
VM221:2 null
*/

Reply:
Is that a feature?

Reply:
Oh!
https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/modules/gamepad/navigator_gamepad.h;l=97;drc=5b8933e94139a0ab5be46141666fdfcce0f624f6
```
  // A reference to the buffer containing the last-received gamepad state. May
  // be nullptr if no data has been received yet.
```

Reply:
Also gamepads remember their slot, a gamepad may not always be present at the first "free" slot.

Reply:
You may want to add a comment for both then inline.

Reply:
Done.